### ISSUE TITLE
privacy.firstparty.isolate true and "Cookies and Site Data" permissions [1556212]

### ISSUE THREAD

**Issue by Dupond** [ghacksuserjs/ghacks-user.js#909]
Hello,

First of all a few informations about this issue (sorry for my bad english, I hope it won't be a problem) : I use Firefox 74.0 on Arch Linux, but I've encountered the problem with previous versions of Firefox too. All extensions (I use only one : uMatrix) have been disabled when testing.

The user.js file says that privacy.firstparty.isolate "may break cross-domain logins and site functionality until perfected", but I'm not sure if it includes the following problem or not : if I keep it set to true as it is in the user.js file, then the cookies for sites set to "Allow" in Preferences > Privacy & Security > Cookies and Site Data > Manage Permissions... are not usable anymore. Is this expected ?

Steps to reproduce :
- go to any website ;
- customize it (I've changed the Preferences of https://searx.tuxcloud.net/ for testing) ;
- allow this website in "Cookies and Site Data > Manage Permissions..." ;
- close Firefox, then reopen it ;
- go back to the allowed website : the preferences are not changed anymore ;
- set "privacy.firstparty.isolate" to "false" : the preferences are now changed.

If this is expected, maybe a note saying that "It's not possible to use Manage Permissions for cookies" would be nice ?

**Comment by Thorin-Oakenpants** [comment]
FPI doesn't affect the UI - that's something else: ~~pb mode~~ (just tested pb mode) / default cookie settings?

Mine isn't disabled. For me, the only disabled part is the "delete cookies and site data when FF closes" - because I have custom settings: cookies are blocked by default and I already clear cookies on close: not sure what combo exactly disables that line.

I'll have to have a play in a vanilla profile, if I get time, to work out exactly what disables that section - TBH I don't ever recall it being disabled

**Comment by Thorin-Oakenpants** [comment]
>  are not usable anymore

My bad - you said usable, not disabled. I **think** the answer is to always add your exceptions via the site permissions panel
- right click in an empty space on the page > page info > permissions
- click the padlock in the urlbar > right arrow > more information > permissions

Most permissions are OA'ed (origin attributed) so it's important that they get added with the `^firstPartyDomain` - if you were manually adding sites to Cookies and Site Data > Manage Permissions, then I wouldn't expect you to know that you need to OA them

see #857 .. pics 2 and 3

**Comment by Dupond** [comment]
Yes ! It works perfectly now, defining the permissions with a simple right click on the page. I would never have figured this out by myself !
Thank you very much, not only for your help, but also for all the time involved in this super great project !

**Comment by Thorin-Oakenpants** [comment]
I'm going to re-open this as a reminder to maybe add something to the user.js / wiki

**Comment by Thorin-Oakenpants** [comment]
It's actually a bug: FPI is meant to fall back to use non-FPI - see [1623104](https://bugzilla.mozilla.org/show_bug.cgi?id=1623104) closed as duplicate of [1556212](https://bugzilla.mozilla.org/show_bug.cgi?id=1556212)
